static/js/containers/ProfilePage_test.js

```diff
@@ -17,106 +17,78 @@ describe("ProfilePage", function() {
   this.timeout(5000);  // eslint-disable-line no-invalid-this
 
   let listenForActions, renderComponent, helper, patchUserProfileStub;
+  let pageUrlStubs = [
+    '/profile/personal',
+    '/profile/education',
+    '/profile/professional',
+    '/profile/privacy'
+  ];
+  let lastPage = pageUrlStubs[pageUrlStubs.length - 1];
+  let prevButtonSelector = '.progress-button.previous';
+  let nextButtonSelector = '.progress-button.next';
+
   beforeEach(() => {
     helper = new IntegrationTestHelper();
     listenForActions = helper.listenForActions.bind(helper);
     renderComponent = helper.renderComponent.bind(helper);
-
     patchUserProfileStub = helper.sandbox.stub(api, 'patchUserProfile');
   });
 
   afterEach(() => {
     helper.cleanup();
   });
 
-  it('marks email_optin and filled_out when saving this page', done => {
-    renderComponent("/profile/privacy").then(([, div]) => {
-      let button = div.querySelectorAll("button")[1];
-      assert.equal(button.innerHTML, "I’m Done!");
+  let confirmSaveButtonBehavior = (updatedProfile, pageElements) => {
+    let { div, button } = pageElements;
+    button = button || div.querySelector(nextButtonSelector);
+    patchUserProfileStub.throws("Invalid arguments");
+    patchUserProfileStub.withArgs(SETTINGS.username, updatedProfile).returns(Promise.resolve(updatedProfile));
+    return listenForActions([
+      REQUEST_PATCH_USER_PROFILE,
+      RECEIVE_PATCH_USER_PROFILE_SUCCESS,
+      START_PROFILE_EDIT,
+      UPDATE_PROFILE_VALIDATION
+    ], () => {
+      TestUtils.Simulate.click(button);
+    });
+  };
+
+  it('navigates backward when Previous button is clicked', () => {
+    let firstPage = pageUrlStubs[0];
+    let secondPage = pageUrlStubs[1];
+    renderComponent(secondPage).then(([, div]) => {
+      let button = div.querySelector(prevButtonSelector);
+      assert.equal(helper.currentLocation.pathname, secondPage);
+      TestUtils.Simulate.click(button);
+      assert.equal(helper.currentLocation.pathname, firstPage);
+    });
+  });
 
+  it('marks email_optin and filled_out when saving ${lastPage}', done => {
+    renderComponent(lastPage).then(([, div]) => {
```

Comment:
:D no worries, the syntax is annoyingly close on the page. backwards compatability...